user_management/ui/views.py

```diff
@@ -0,0 +1,53 @@
+from django.contrib import messages
+from django.contrib.auth import get_user_model
+from django.contrib.auth.tokens import default_token_generator
+from django.core.exceptions import PermissionDenied
+from django.http import Http404
+from django.utils.encoding import force_text
+from django.utils.http import urlsafe_base64_decode
+from django.views import generic
+
+
+class VerifyUserEmailView(generic.RedirectView):
+    """
+    A view which verifies a user's email address.
+
+    Accessed via a link in an email sent to the user, which contains both a
+    uid generated from the user's pk, and a token also generated from the user
+    object, for verification.  If everything lines up, it makes the user
+    active.
```

Comment:
https://github.com/django/django/blob/1.8.8/django/contrib/auth/tokens.py#L46 apparently it doesn't, but it's specific to the particular user object, so it effectively does tie to the email address the user entered.  (You can't change your email address without logging in, so I don't think that exposes any security holes.)